stackslib/src/chainstate/stacks/boot/miners.clar

```diff
@@ -0,0 +1,21 @@
+;; This contract governs a StackerDB instance in which the current and previous
+;; miner can send their blocks to Stackers for an aggregate signature.
+;; This is a placeholder smart contract, which allows the node to advertize
+;; that it replicates the state for this StackerDB while maintaining the power
+;; to generate the config and signer slots directly.
+
+;; StackerDB-required method to get the allocation of slots for signers.
+;; The values here are ignored.
+(define-public (stackerdb-get-signer-slots)
+    (ok (list )))
+
+;; StackerDB-required method to get the DB configuration.
+;; The values here are ignored.
+(define-public (stackerdb-get-config)
+    (ok {
+        chunk-size: u0,
+        write-freq: u0,
+        max-writes: u0,
+        max-neighbors: u0,
+        hint-replicas: (list )
+    }))
```

Comment:
Why is it necessary for this contract to exist at all?

Reply:
I think it will werror out when "subscribing" to the stacker db instance if there is no contract found, but I can see if that is actually the case.

Reply:
The contract has to exist so the RPC endpoints will work.  Normally (i.e. for every other stacker DB), these methods are only evaluated every Bitcoin block in order to generate a (node-side) StackerDB configuration.  In the specific case of `.miners`, the node generates that configuration on its own.

Reply:
Does it make more sense to special case the RPC endpoints for `.miners`? The config lookups are already treated as special cases, so its not like this really avoids the need for special casing, and having a contract around that returns data that doesn't correspond to the actual behavior of the StackerDB seems confusing if it can be avoided.

Reply:
We already expect users to understand that any contracts instantiated with the boot address behave differently than expected, so I don't think it's unreasonable to expect the same understanding for `.miners`.  Also, there are currently four StackerDB-specific endpoints, and some others (e.g. /v2/neighbors) that convey StackerDB data.  I'd rather not have to duplicate them all just for `.miners`.

Reply:
Why would you need to duplicate the whole endpoints? What fails in those endpoints if the contract doesn't exist? It's true that boot address contracts do behave somewhat differently than normal contracts, but none of them actually return incorrect data (as this one would).

Reply:
> Why would you need to duplicate the whole endpoints?

It sounded like that's what you're asking for?  A set of API endpoints specific to querying, storing, and getting the metadata for chunks of data written for `.miners`?

> What fails in those endpoints if the contract doesn't exist?

The StackerDB API endpoints query the `stackerdbs.sqlite` database directly; they don't query the Clarity VM.  Instead, the `PeerNetwork` will do so whenever it detects that the burnchain has advanced -- it will re-evaluate these two Clarity functions to synthesize a `StackerDBConfig`, and will then update the `stackerdbs.sqlite` database with it (this is documented in `src/stackslib/net/stackerdb/mod.rs`).

If the contract doesn't exist -- i.e. if `PeerNetwork` doesn't synthesize a `StackerDBConfig` struct for its StackerDB replica -- then there won't be any configuration data for them in `stackerdbs.sqlite`, and thus the endpoint will fail.

> but none of them actually return incorrect data (as this one would).

The RPC endpoints don't query the smart contract for any state (as described above, they're only involved in the `StackerDBConfig` generation), so this shouldn't be an issue.

Reply:
> It sounded like that's what you're asking for? A set of API endpoints specific to querying, storing, and getting the metadata for chunks of data written for .miners?

I see -- no, that's not what I was suggesting. I was suggesting adding a check anywhere that would fail without the presence of a contract for `.miners`.

> The RPC endpoints don't query the smart contract for any state (as described above, they're only involved in the StackerDBConfig generation), so this shouldn't be an issue.

Yep, I see that now, which still leaves me wondering why the `.miners` contract needs to exist.

> If the contract doesn't exist -- i.e. if PeerNetwork doesn't synthesize a StackerDBConfig struct for its StackerDB replica -- then there won't be any configuration data for them in stackerdbs.sqlite, and thus the endpoint will fail.

But this is exactly what this PR _avoids_ for `.miners`, right? `create_or_reconfigure_stackerdbs` has a special case for `.miners`, and never actually invokes the Clarity VM. So, it doesn't seem to need the contract to exist either.

Applying this diff to the PR, for example, still leaves the end-to-end test passing -- meaning that the stackerdbs instance still initializes the miners stackerdb, still accepts the block proposal, and still returns it, all without `.miners` ever existing. If its actually the case that the contract doesn't need to exist for the stackerdb to work, I'd much prefer not instantiating it at all. As you say, the contract doesn't respond on those public functions with data that actually corresponds to anything.

```diff
diff --git a/stackslib/src/clarity_vm/clarity.rs b/stackslib/src/clarity_vm/clarity.rs
index 9a18decfd..8bbd1cb95 100644
--- a/stackslib/src/clarity_vm/clarity.rs
+++ b/stackslib/src/clarity_vm/clarity.rs
@@ -1288,28 +1288,6 @@ impl<'a, 'b> ClarityBlockConnection<'a, 'b> {
             let miners_contract_tx =
                 StacksTransaction::new(tx_version.clone(), boot_code_auth.clone(), payload);
 
-            let miners_initialization_receipt = self.as_transaction(|tx_conn| {
-                // initialize with a synthetic transaction
-                debug!("Instantiate {} contract", &miners_contract_id);
-                let receipt = StacksChainState::process_transaction_payload(
-                    tx_conn,
-                    &miners_contract_tx,
-                    &boot_code_account,
-                    ASTRules::PrecheckSize,
-                )
-                .expect("FATAL: Failed to process .miners contract initialization");
-                receipt
-            });
-
-            if miners_initialization_receipt.result != Value::okay_true()
-                || miners_initialization_receipt.post_condition_aborted
-            {
-                panic!(
-                    "FATAL: Failure processing .miners contract initialization: {:#?}",
-                    &miners_initialization_receipt
-                );
-            }
-
             /////////////////// .pox-4 ////////////////////////
             let first_block_height = self.burn_state_db.get_burn_start_height();
             let pox_prepare_length = self.burn_state_db.get_pox_prepare_length();
```

Reply:
@jcnelson if nothing breaks to remove the .miners contract, I am happy to do so. I thought you mentioned that the RPC calls would fail though D:

Reply:
I thought they would, but it turns out I was wrong -- the RPC endpoint only looks at the `stackerdbs.sqlite` DB to service queries, not the smart contract in the chainstate.  So as long as the `StackerDBConfig` is getting synthesized and applied, then the RPC endpoints should work.